spec/rubocop/cop/rspec/sort_metadata_spec.rb

```diff
@@ -0,0 +1,222 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::SortMetadata do
+  it 'does not register an offense when using only symbol metadata ' \
+     'in alphabetical order' do
+    expect_no_offenses(<<~RUBY)
+      RSpec.describe 'Something', :a, :b do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using only symbol metadata, ' \
+     'but not in alphabetical order' do
+    expect_offense(<<~RUBY)
+      RSpec.describe 'Something', :b, :a do
+                                  ^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      RSpec.describe 'Something', :a, :b do
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when using only a hash of metadata ' \
+     'with keys in alphabetical order' do
+    expect_no_offenses(<<~RUBY)
+      context 'Something', baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using only a hash of metadata, ' \
+     'but with keys not in alphabetical order' do
+    expect_offense(<<~RUBY)
+      context 'Something', foo: 'bar', baz: true do
+                           ^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      context 'Something', baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when using mixed metadata ' \
+     'in alphabetical order (respectively)' do
+    expect_no_offenses(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata, ' \
+     'but only the hash keys are in alphabetical order' do
+    expect_offense(<<~RUBY)
+      it 'Something', :b, :a, baz: true, foo: 'bar' do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata, ' \
+     'but only the symbol keys are in alphabetical order' do
+    expect_offense(<<~RUBY)
+      it 'Something', :a, :b, foo: 'bar', baz: true do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata ' \
```

Comment:
Aren't we missing an example that we sort them independently, even though  a purely alphabetical order will suggest that the symbols metadata should be last, like:

```
expect_offense(<<~RUBY)
       it 'Something', :c, :z, foo: 'bar', baz: true do
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
       end
     RUBY
```

Reply:
Actually [this ones](https://github.com/rubocop/rubocop-rspec/pull/1378/files#diff-4f710edcf279102cf88580e810d52f474333e8a181c14ed2592e854ae7a54c86R130) covers the case